meson.build

```diff
@@ -0,0 +1,624 @@
+
+	if have == true
+		conf.set10('HAVE_' + decl.underscorify().to_upper(), have)
+	endif
+endforeach
+
+foreach ident : [
+        ['memfd_create',      '''#include <sys/mman.h>'''],
+        ['gettid',            '''#include <sys/types.h>
+                                 #include <unistd.h>'''],
+        ['pivot_root',        '''#include <stdlib.h>
+                                 #include <unistd.h>'''],     # no known header declares pivot_root
+        ['setns',             '''#include <sched.h>'''],
+        ['renameat2',         '''#include <stdio.h>
+                                 #include <fcntl.h>'''],
+        ['kcmp',              '''#include <linux/kcmp.h>'''],
+        ['keyctl',            '''#include <sys/types.h>
+                                 #include <keyutils.h>'''],
+        ['bpf',               '''#include <sys/syscall.h>
+                                 #include <unistd.h>'''],
+        ['statx',             '''#include <sys/types.h>
+                                 #include <sys/stat.h>
+                                 #include <unistd.h>'''],
+        ['pidfd_send_signal', '''#include <stdlib.h>
+                                 #include <unistd.h>
+                                 #include <signal.h>
+                                 #include <sys/wait.h>'''],
+        ['pidfd_open',        '''#include <stdlib.h>
+                                 #include <unistd.h>
+                                 #include <signal.h>
+                                 #include <sys/wait.h>'''],
+        ['execveat',          '''#include <unistd.h>'''],
+        ['close_range',       '''#include <unistd.h>'''],
+        ['mount_setattr',     '''#include <sys/mount.h>'''],
+        ['move_mount',        '''#include <sys/mount.h>'''],
+        ['open_tree',         '''#include <sys/mount.h>'''],
+        ['strlcpy',	      '''#include <string.h>'''],
+        ['strlcat',	      '''#include <string.h>'''],
+        ['sethostname',	      '''#include <unistd.h>'''],
+        ['faccessat',         '''#include <fcntl.h>
+                                 #include <unistd.h>'''],
+        ['unshare',	      '''#include <sched.h>'''],
+        ['prlimit',	      '''#include <sys/time.h>
+                               #include <sys/resource.h>'''],
+        ['prlimit64',	      '''#include <sys/time.h>
+                               #include <sys/resource.h>'''],
+]
+
+        have = cc.has_function(ident[0], prefix : ident[1], args : '-D_GNU_SOURCE')
+	if have == true
+		conf.set10('HAVE_' + ident[0].to_upper(), have)
+	endif
+endforeach
+
+threads = dependency('threads')
+libseccomp = cc.find_library('seccomp', required : false)
+if libseccomp.found()
+	conf.set10('HAVE_SECCOMP', libseccomp.found())
+
+	seccomp_headers = '''
+	#include <seccomp.h>
+	'''
+
+	if cc.has_function('seccomp_notify_fd', prefix : seccomp_headers, args : '-D_GNU_SOURCE', dependencies: libseccomp)
```

Comment:
libseccomp comes with a pkg-config file, so you can just use

```
libseccomp = dependency('libseccomp')
if libseccomp.version().version_compare('>=2.5.0')
    # https://github.com/seccomp/libseccomp/commit/dead12bc788b259b148cc4d93b970ef0bd602b1a
    conf.set10('HAVE_DECL_SECCOMP_NOTIFY_FD', true)
endif
if libseccomp.version().version_compare('>=2.0.0')
    # https://github.com/seccomp/libseccomp/commit/6220c8c0fc479d97b6d3e3166a4e46fbfe25a3c0
    conf.set10('HAVE_DECL_SECCOMP_SYSCALL_RESOLVE_NAME_ARCH', true)
endif
```

Reply:
I've simplified it a bit at least.